src/Microsoft.IdentityModel.Logging/LogHelper.cs

```diff
@@ -262,10 +262,10 @@ public static Exception LogExceptionMessage(Exception exception)
         public static Exception LogExceptionMessage(EventLevel eventLevel, Exception exception)
         {
             if (exception == null)
-                return null;
+                return null!;
```

Comment:
this doesn't seem right, why not change the return type to `Exception?`

Reply:
You need to return null here, so adding `null!` is not correct because we are returning null. Only add `!` when you know for sure the value will never be null.

Instead, note that the return type can be null instead:
`public static Exception? LogExceptionMessage(EventLevel eventLevel, Exception exception)`

and bubble it up:
`public static Exception? LogExceptionMessage(Exception exception)`

Reply:
My mistake, I'll update this thanks @keegan-caruso & @jennyf19

Reply:
why not `exception = exception ?? throw new ArgumentNullException(nameof(exception));` for example?

Reply:
returning null is still incorrect.

I would argue that passing null is a usage error of the API.

It would be a behavioral breaking change, but what about throwing an argument null exception if passed null? We would need to include it in the breaking change list, but it is targeting the 7x branch.

Thoughts @jennyf19  and @FuPingFranco ?

Reply:
@keegan-caruso i agree with you about return null, that seems to be a programming mistake and can lead to a null ref at any time.

How about we add a debug assert if the exception passed in is null and return new ArgumentNullException("null exception was passed to logger");

@jennyf19  not sure where you are suggesting we have this code.
why not exception = exception ?? throw new ArgumentNullException(nameof(exception)); for example?